111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,236 @@
+Because of this fact and also potential security risk, we decided to reject this alternative.
+
+### Using fork of Kaniko
+
+As the Kaniko was archived, it was then [forked by Chainguard](https://github.com/chainguard-dev/kaniko).
+Currently, it's maintained, but they are not providing any image - which means that we would have to build our own.
+It's not a blocker for us, but it would mean some kind of maintenance to be done on our side.
+The main reason why I rejected this alternative is, that in their [README](https://github.com/chainguard-dev/kaniko?tab=readme-ov-file#history-and-status) they mention that "If another active and community-supported fork emerges, we'll happily shut this one down and migrate to that".
+Which means for us that we would have to migrate to something else possibly in few months again, and maybe doing this proposal process once again.
+Having something that is supported, maintained, and used widely is better alternative for us currently.
+
+### Forking and maintaining Kaniko repository on our own
+
+Another alternative was to fork the Kaniko repository and maintain it on our own.
+Even if we wouldn't add any new features to it, it's still a maintenance burden - we would have to fix bugs and CVEs that may occur during the time.
+Also, we would then have to maintain it for longer time - in case that there will be non-Strimzi (but also Strimzi) users using it in their own builds.
+That would mean that even if we for example deprecate and remove the Connect Build feature, we would still have to maintain the Kaniko repository for the other users.
+Finally - it's written in Go and based on our previous experience with Go component (Canary), our team doesn't have enough knowledge of Go in order to maintain it fully.
+
+### Removing Connect Build feature completely
+
+Even removing the Connect Build feature completely was a possible alternative.
+But based on the discussions on Slack or GitHub, and also issues on GitHub, it's still quite often used by our Strimzi users.
+So removing it now when the alternative - Image Volumes - is not used widely and supported by all Kubernetes versions Strimzi supports, is not a good idea.
+
+### Using Image Volumes feature in Kubernetes only
+
+The alternative of using Image Volumes feature is connected to the previous section.
+As I already mentioned, the Image Volumes feature is not widespread, and it's not supported by all Kubernetes versions Strimzi supports.
+Even if it's possible solution for the future, we need to wait until it's used and supported widely before moving to it.
+
+### Using other tools like BuildKit
+
+During implementing and testing the PoC with Buildah, I tried also different tools like BuildKit.
+BuildKit is mainly rootful, but it can be used rootless.
+However, in order to build an image inside the container, it needs some hacking and workarounds, which are not that straighforward as in case of Buildah.
```

Comment:
```suggestion
However, in order to build an image inside the container, it needs some hacking and workarounds, which are not that straightforward as in case of Buildah.
```